crucible-mir/src/Mir/TransTy.hs

```diff
@@ -1240,6 +1240,47 @@ tupleFieldRef tys i tpr ref = do
     ref' <- mirRef_agElem_constOffset (fromIntegral i) 1 valTpr ref
     return $ MirPlace valTpr ref' NoMeta
 
+-- | Acquire a reference to the union field specified by the given index and
+-- type. The field type must match the type specified in the provided `M.Adt`.
+--
+-- Additionally, the field type must match the expression type with which the
+-- union was initialized - see Note [union representation] in `Mir.Trans`.
```

Comment:
Haddock nit pick: Haddock uses double quotes to hyperlink module names, not backticks (which are for identifiers):

```diff
--- union was initialized - see Note [union representation] in `Mir.Trans`.
+-- union was initialized - see Note [union representation] in "Mir.Trans".
```

Reply:
Sigh - one day, I'll get Haddock right on the first try.

I moved the note to `TransTy`, so this reference to `Mir.Trans` was removed entirely.